tests/integration/relation_tests/helpers.py

```diff
@@ -1,35 +1,64 @@
 #!/usr/bin/env python3
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
-import logging
+import json
 from typing import Optional
 
 import yaml
 from pytest_operator.plugin import OpsTest
 
-logger = logging.getLogger(__name__)
-
 
 async def get_application_relation_data(
-    ops_test: OpsTest, application_name: str, key: str
+    ops_test: OpsTest,
+    application_name: str,
+    relation_name: str,
+    key: str,
+    relation_id: str = None,
+    relation_alias: str = None,
 ) -> Optional[str]:
     """Get relation data for an application.
 
     Args:
         ops_test: The ops test framework instance
         application_name: The name of the application
+        relation_name: name of the relation to get connection data from
         key: key of data to be retrieved
-
+        relation_id: id of the relation to get connection data from
+        relation_alias: alias of the relation (like a connection name)
+            to get connection data from
     Returns:
         the that that was requested or None
             if no data in the relation
-
     Raises:
-        ValueError if it's not possible to get application unit data.
+        ValueError if it's not possible to get application unit data
+            or if there is no data for the particular relation endpoint
+            and/or alias.
     """
     unit_name = f"{application_name}/0"
     raw_data = (await ops_test.juju("show-unit", unit_name))[1]
+
     if not raw_data:
         raise ValueError(f"no unit info could be grabbed for {unit_name}")
     data = yaml.safe_load(raw_data)
-    return data[unit_name]["relation-info"][0]["application-data"].get(key)
+
+    # Filter the data based on the relation name.
+    relation_data = [v for v in data[unit_name]["relation-info"] if v["endpoint"] == relation_name]
```

Comment:
I'm a little confused about the meaning of `v` - does it just mean `variable`?

Reply:
https://www.w3schools.com/python/python_lists_comprehension.asp :)